static/js/panel.js

```diff
@@ -6,9 +6,42 @@ var Panel = new (class Panel {
     this.content = document.getElementById("panel-content");
     this.accelEnabledCheckbox = document.getElementById("panel-accel-enable");
 
-    this.permalinkNode = this.findLink("Permalink");
-    this.logNode = this.findLink("Log");
-    this.rawNode = this.findLink("Raw");
+    this.permalinkNode = this.findItem("Permalink");
+    this.logNode = this.findItem("Log");
+    this.rawNode = this.findItem("Raw");
+
+    this.markdown = {
+      "filename": {
+        node: this.findItem("Filename Link"),
+        isEnabled: () => {
+          return Highlight?.selectedLines.size > 0;
```

Comment:
nit: This one can probably be unconditionally enabled? I could see it being useful without any selected lines.